src/python/pants/backend/python/goals/setup_py.py

```diff
@@ -490,6 +502,28 @@ async def generate_chroot(request: SetupPyChrootRequest) -> SetupPyChroot:
     target = exported_target.target
     resolved_setup_kwargs = await Get(SetupKwargs, ExportedTarget, exported_target)
     setup_kwargs = resolved_setup_kwargs.kwargs.copy()
+
+    setup_script = setup_kwargs.pop("setup_script", None)
+    if setup_script:
+        # The target points to an existing setup.py script, so use that.
+        invalid_keys = set(setup_kwargs.keys()) - {"name", "version"}
+        if invalid_keys:
+            raise InvalidSetupPyArgs(
+                f"The `provides` field in {exported_addr} specifies a setup_script kwarg, so it "
+                f"must only specify the name and version kwargs, but it also specified "
+                f"{','.join(sorted(invalid_keys))}."
+            )
+        stripped_setup_script = await Get(
+            StrippedSourceFileNames, SourcesPaths(files=(setup_script,), dirs=())
+        )
+        return SetupPyChroot(
+            sources.digest,
```

Comment:
This might not have the `setup.py`, unless it was contained by a `python_library` that is in the dependencies of the python_distribution.

Instead, I think you want to get the `Digest` for this glob and use `MergeDigests`.

Reply:
My intention was that you point to a setup.py that is in the deps, by design. I don't want to start letting you omit it from your regular python_library .